packages/app-mobile/utils/checkPermissions.ts

```diff
@@ -1,25 +1,46 @@
+import { _ } from '@joplin/lib/locale';
+import shim from '@joplin/lib/shim';
 import Logger from '@joplin/utils/Logger';
 
-const { Platform, PermissionsAndroid } = require('react-native');
+import { Platform, PermissionsAndroid, Permission } from 'react-native';
 const logger = Logger.create('checkPermissions');
 
-type rationale = {
+type Rationale = {
 	title: string;
 	message: string;
 	buttonPositive?: string;
 	buttonNegative?: string;
 	buttonNeutral?: string;
 };
 
-export default async (permissions: string, rationale?: rationale) => {
+interface ConfirmMessage {
+	title: string;
+	message: string;
+}
+
+interface Options {
+	rationale?: Rationale;
+	confirmMessage?: ConfirmMessage;
+}
+
+export default async (permissions: Permission, { rationale, confirmMessage }: Options = {}) => {
 	// On iOS, permissions are prompted for by the system, so here we assume it's granted.
 	if (Platform.OS !== 'android') return PermissionsAndroid.RESULTS.GRANTED;
 
-	let result = await PermissionsAndroid.check(permissions);
-	logger.info('Checked permission:', result);
-	if (result !== PermissionsAndroid.RESULTS.GRANTED) {
-		result = await PermissionsAndroid.request(permissions, rationale);
+	const granted = await PermissionsAndroid.check(permissions);
+	logger.info('Checked permission:', granted);
+	if (granted) {
```

Comment:
Note: The return type of `PermissionsAndroid.check` is a `boolean`, not a string. As a result, the `result !== PermissionsAndroid.RESULTS.GRANTED` was always `true`. Presumably, [`PermissionsAndroid.request`](https://reactnative.dev/docs/0.79/permissionsandroid#request) does nothing if called when permission is already granted.